pkg/printers/internalversion/printers.go

```diff
@@ -167,6 +167,7 @@ func AddHandlers(h printers.PrintHandler) {
 
 	jobColumnDefinitions := []metav1.TableColumnDefinition{
 		{Name: "Name", Type: "string", Format: "name", Description: metav1.ObjectMeta{}.SwaggerDoc()["name"]},
+		{Name: "Suspend", Type: "boolean", Description: batchv1.JobSpec{}.SwaggerDoc()["suspend"]},
```

Comment:
I'm now leaning more towards the approach as we have for pods where we have the `STATUS` column reflecting the current pod state, doing lookup to multiple places: https://github.com/kubernetes/kubernetes/blob/master/pkg/printers/internalversion/printers.go#L834-L959.

The states I can we could have for Job:
- `Complete` - when has the `Complete=True` condition
- `Failed` - when has the `Failed=True` condition
- `FailureTarget` - when has the `FailureTarget=True` condition
- `Suspended` - when has the `Suspended=True` condition, and`.spec.suspend=true`
- `Resuming` - when has the `Suspended=True` condition, but `.spec.suspend != true`
- `Suspending` - when has the `.spec.suspend=true`, but no `Suspended=True` condition
- `Terminating` - when `deletionTimestamp != nil`
- `Running` - none of the above, potentially maybe when `status.startTime != nil` (and the we could have `Starting` when `.status.startTime=nil`.

WDYT @kannon92 @alculquicondor @soltysh ?

Reply:
I wouldn't make a distinction between suspended, resuming, suspending. I would just use the condition.
Other than that, SGTM

Reply:
I like the change suggested by @mimowo.

Reply:
I'm ok either with dropping `Resuming` and `Suspending` or keeping them. 

Regardless, we should set the state based on condition as the actual state, not the spec.suspend (desired state).

Reply:
> I rather prefer `Suspended` here as it conveys better the **state** of the job. I find using the verb `Suspend` a bit confusing

I originally had it as "Suspended", but then I saw that CronJob's suspend is shown as "Suspend", so I changed it to "Suspend" for consistency (emphasis on line 185):
https://github.com/kubernetes/kubernetes/blob/04559e6edae2ac0b79955063b5bbcb2d5a242947/pkg/printers/internalversion/printers.go#L181-L185

Re: dropping the new column and expanding Status' behavior, have we reached an agreement? Should I go ahead with the changes?

Reply:
I think @mimowo 's suggestion is a great idea since is coherent with the UX of other objects. I'd go with that

Reply:
Do you suggest closing this PR, and starting a new one?

Do we have a consensus on keeping or removing `Suspending` and `Resuming`?

Reply:
> I originally had it as "Suspended", but then I saw that CronJob's suspend is shown as "Suspend",

I see, but here are some differences with CronJob: I don't think CronJob supports conditions, can it be "finished"? I feel since job can be in multiple "states" it makes sense to put them into a single column to avoid creating a column per field or condition. Similarly as for pod.

> Do you suggest closing this PR, and starting a new one?

I would suggest updating this one. You can put a new PR so that we don't lose history in case some changes get reverted due to new insights, but I think keeping the PR open is beneficial to also keep discussions in one place, because they would be similar in the new PR.

> Do we have a consensus on keeping or removing Suspending and Resuming?

The transient states `resuming` and `suspending` are very short, so I'm ok either way.

Reply:
@mimowo, @kaisoz, @kannon92, @alculquicondor PTAL @ #123349 which supersedes this PR.

Reply:
You can add commit's to this PR.

Reply:
Sounds good, I'll close the other one.